src/Umbraco.Cms.Integrations.Search.Algolia/Services/AlgoliaSearchPropertyIndexValueFactory.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Text.Json;
+
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Integrations.Search.Algolia.Services
+{
+    public class AlgoliaSearchPropertyIndexValueFactory : IAlgoliaSearchPropertyIndexValueFactory
```

Comment:
I've not quite followed how this class works with the `ParserService` you've also defined in the library project.  It seems to be doing a similar thing and has some duplicated code from there.

Are they two versions of work doing similar things?  If so, think we should try to combine so there's only one class responsible got "getting property values for the index".